springserve/_reporting.py

```diff
@@ -3,34 +3,104 @@
 
 from datetime import datetime
 
+from time import sleep
+
 import pandas
 
+from link import lnk
+
+msg = lnk.msg
+
 class _ReportingResponse(_VDAPIMultiResponse):
-    
+
```

Comment:
You should create a constructor here that creates the self.dataframe variable.  It is only ever used in this class.